test/functional/web/safari-nativewebtap-e2e-specs.js

```diff
@@ -1,270 +1,234 @@
-//           await driver.execute('mobile: scroll', {direction: 'down'});
-
-//           await driver.elementByLinkText(PAGE_3_LINK).click();
-
-//           await spinTitleEquals(driver, PAGE_3_TITLE, SPIN_RETRIES);
-//         });
-
-//         it('should be able to tap on a button', async function () {
-//           this.retries(5);
-
-//           await loadPage(driver, GUINEA_PIG_PAGE);
-
-//           (await driver.source()).should.not.include('Your comments: Hello');
-
-//           await driver.elementByName('comments').type('Hello');
-
-//           await driver.elementByName('submit').click();
-
-//           await retryInterval(5, 500, async function () {
-//             const src = await driver.source();
-//             return src.should.include('Your comments: Hello');
-//           });
-//         });
-
-//         it('should be able to handle an alert', async function () {
-//           await loadPage(driver, GUINEA_PIG_PAGE);
-
-//           await driver.elementById('alert1').click();
-//           await retryInterval(5, 1000, driver.acceptAlert.bind(driver));
-//           await driver.title().should.eventually.include('I am a page title');
-//         });
-
-//         describe('with tabs -', function () {
-//           before(async function () {
-//             if (skipped || !deviceName.toLowerCase().includes('ipad')) {
-//               return this.skip();
-//             }
-//             await loadPage(driver, GUINEA_PIG_PAGE);
-
-//             // open a new tab and go to it
-//             await driver.elementByLinkText('i am a new window link').click();
-
-//             await retryInterval(10, 1000, async function () {
-//               await driver.title().should.eventually.eql('I am another page title');
-//             });
-//           });
-
-//           it('should be able to tap on an element', async function () {
-//             await loadPage(driver, GUINEA_PIG_PAGE);
-
-//             await driver.elementByLinkText(PAGE_3_LINK).click();
-
-//             await spinTitleEquals(driver, PAGE_3_TITLE, SPIN_RETRIES);
-
-//             await driver.back();
-
-//             // try again, just to make sure
-//             await driver.elementByLinkText(PAGE_3_LINK).click();
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import _ from 'lodash';
+import { initSession, deleteSession, hasDefaultPrebuiltWDA, MOCHA_TIMEOUT } from '../helpers/session';
+import { amendCapabilities, SAFARI_CAPS, DEVICE_NAME, DEVICE_NAME_FOR_SAFARI_IPAD } from '../desired';
+import {
+  openPage, spinTitleEquals, spinTitle, GUINEA_PIG_PAGE,
+  GUINEA_PIG_SCROLLABLE_PAGE, GUINEA_PIG_APP_BANNER_PAGE
+} from './helpers';
+import { retryInterval } from 'asyncbox';
+import B from 'bluebird';
+
+chai.should();
+chai.use(chaiAsPromised);
+
+const caps = amendCapabilities(SAFARI_CAPS, {
+  'appium:safariInitialUrl': GUINEA_PIG_PAGE,
+  'appium:nativeWebTap': true,
+  'appium:fullReset': true,
+  'appium:noReset': false,
```

Comment:
I'm not sure if this is an appropriate way to avoid ci tests fail (I'd like to just close all tabs in mobile safari before running these tests)

Reply:
there is a possibility to influence global safari settings in Simulator: https://appium.github.io/appium-xcuitest-driver/4.16/execute-methods/#mobile-updatesafaripreferences

Maybe this could help.
Enabling fullReset may significantly slow down tests as it resets the whole simulator

Reply:
sorry to be late, so far I haven't managed to find a way to close tabs via `mobile: updateSafariPreferences`. To me it looks infeasible. 

If no problem, I'm thinking about trying to open Preferences app and close the safari tabs via GUI at `before` mocha function.